apps/next-product-starter/src/pages/video/[slug].tsx

```diff
@@ -70,9 +74,17 @@ const PlayerPage = ({resource}: any) => {
   const fullscreenWrapperRef = React.useRef<HTMLDivElement>(null)
 
   React.useEffect(() => {
-    // @ts-ignore
-    videoService.send('LOAD_RESOURCE', {resource: currentResource})
-  }, [])
+    setCurrentResource(resource)
+    videoService.send({type: 'LOAD_RESOURCE', resource: currentResource})
+
+    const delayAutoAdvance = setTimeout(() => {
+      if (getPlayerPrefs().autoplay) {
+        videoService.send({type: 'PLAY'})
+      }
+    }, 10)
+
+    return () => clearTimeout(delayAutoAdvance)
```

Comment:
this doesn't seem like the best approach, but it's the only I could think of that worked. the issue here is with `videoElemRef` not being registered right away (and additional checks using machine context aren't helping either), resulting in machine transition into playing state with actual video being paused. it's a strange one.